### PR TITLE
Release v2.2.0

### LINKED ISSUE #350: Releasing v2.2.0
After https://github.com/spglib/spglib/pull/349 is merged, I would like to release the next version possibly v2.2.0.
I think the corrections of Hall symbols (https://github.com/spglib/spglib/pull/317) deserve minor change (v2.2.0), not patch change (v.2.1.1).
@LecrisUT @atztogo do you have any ongoing PRs to be included in the next version?
If not, I will write a release note and a corresponding PR so far.

### Remained TODOs
- [x] https://github.com/spglib/spglib/pull/357
- [x] https://github.com/spglib/spglib/pull/371

### PR DESCRIPTION
**Pull request by lan496**
Closes https://github.com/spglib/spglib/issues/350

After this PR is merged, I'll put `v2.2.0` tag to the commit.

List of merged PRs were generated by
```
gh pr list --state merged --search "merged:>2023-09-09" --json url,title,number | jq -r '.[] | "[[#\(.number)]](\(.url)) - \(.title)"'
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "scikit_build_core.build"
 
 [project]
 name = "spglib"
-version = "2.1.0"
+version = "2.2.0"
 requires-python = ">=3.7"
```
Comment (atztogo) [inline pyproject.toml:8]: python 3.7 is EOL. Should we change it to 3.8? For me, either is fine.

**Review by atztogo — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (LecrisUT) [reply pyproject.toml:8]: Agree should drop `3.7`. Let's do it the first thing after the release

**Review by LecrisUT — APPROVED** [review]

**Review by LecrisUT — CHANGES_REQUESTED** [review]
Also update the version in `CMakeLists.txt`

**Review by LecrisUT — APPROVED** [review]

**Comment by lan496** [comment]
I also agree with @LecrisUT. Let's consider to drop Python 3.7 after this release.
https://github.com/spglib/spglib/issues/373

**Comment by lan496** [comment]
I will merge this PR and release `v2.2.0` tomorrow.

**Comment by atztogo** [comment]
> I will merge this PR and release `v2.2.0` tomorrow.

Thanks @lan496!

**Comment by lan496** [comment]
```
git checkout develop
git pull upstream develop
git tag v2.2.0
git push upstream v2.2.0
```

**Comment by lan496** [comment]
Oops, I forgot to revert sphinx-book-theme dependency.
https://github.com/spglib/spglib/actions/runs/7136063269/job/19434451228

**Comment by LecrisUT** [comment]
Oh you're right, forgot about that as well. There is a pre-release out, I can patch the rtd to get it working when I get to work.

Do you want to move the tag after that?

**Comment by lan496** [comment]
🙏

**Comment by lan496** [comment]
For now, I will create a new branch to release v2.2.0 and revert the commit.

**Comment by lan496** [comment]
```
gco -b release-v2.2.0
git revert 16a1efd
git tag v2.2.0 -f
git push upstream release-v2.2.0
git push upstream v2.2.0 -f
```

**Comment by LecrisUT** [comment]
I was concerned that the documentation would not build (and it doesn't), but apparently we caught a bug/race-condition in rtd that it still used the old commit

https://readthedocs.org/projects/spglib/builds/22790068/

We can count our blessings and leave it at this, or we force push the tag again

PS: No need for a new branch for this since main should have this commit as well. But naming convention should be `release-v2.2` can still rename the branch since the tag is independent of that, or better yet, delete after we have that commit in `develop`, until we find a need for patching/releasing `v2.2.1`

**Comment by lan496** [comment]
I've renamed the branch to `release-v2.2`
image

I'm making PR to merge `release-v2.2` into `develop`.

**Comment by LecrisUT** [comment]
> I'm making PR to merge `release-v2.2` into `develop`.

Let me take over that, I'll add the fixes to rtd there as well

**Comment by lan496** [comment]
@LecrisUT Thanks. I defer to you for applying patch on top of `release-v2.2`. Then, let's release v2.2.1 quickly.

**Comment by LecrisUT** [comment]
No need for a `v2.2.1` since this only affects CI

**Pull request merged**